packages/studio-be/src/core/migration/bot-migration-service.ts

```diff
@@ -68,58 +58,66 @@ export class BotMigrationService {
     await this.bpfs.forBot(botId).upsertFile('/', 'migrations.json', JSON.stringify(entries, undefined, 2))
   }
 
-  private displayMigrationStatus(configVersion: string, missingMigrations: MigrationFile[], logger: sdk.Logger) {
-    const migrations = missingMigrations.map(x => this.migService.loadedMigrations[x.filename].info)
+  private async getMissingMigrations(botId: string, migrations: MigrationFile[]): Promise<MigrationFile[]> {
+    const botConfig = await this.configProvider.getBotConfig(botId)
+    const opts = await this.migService.getMigrationOpts({ botId, botConfig, isDryRun: true })
+
+    const missing = await Promise.mapSeries(migrations, async migration => {
+      try {
+        const result = await this.migService.loadedMigrations[migration.filename].up(opts)
+        if (result.hasChanges) {
+          return migration
+        }
+      } catch (err) {
+        return migration
```

Comment:
So if the migration fails, we want to re-run it?

Reply:
I'm not so sure about that, technically at this point we're at the dry run, i'm not sure how it should be handled, do we just exit the whole process ? just skip the migration? keep the bot unmounted?

Reply:
Changed the behavior. It will display the exception / details and skip that migration. Preventing the mount or exiting the process would be a bit too cumbersome to handle.

And basically, we'll catch those errors in development